### PR TITLE
Use the new fil_actor_interface crate

### LINKED ISSUE #2628: Move `actor-interface` to `fil-actor-states`
**Issue summary**

The `forest_actor_interface` no longer depends on any Forest specific logic and should be moved into the same repository as the actor crates.

- [x] Rename `forest_actor_interface` to `fil_actor_interface`.
- [x] Move crate to `fil-actor-states`.
- [x] Update Forest to use the new `fil_actor_interface` crate.

**Other information and links**

https://github.com/ChainSafe/fil-actor-states

### PR DESCRIPTION
**Pull request by sudo-shashank**
## Summary of changes

Changes introduced in this pull request:
- Updated Forest to use the new fil_actor_interface crate.

## Reference issue to close (if applicable)

Closes #2628 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

File: utils/forest_shim/src/lib.rs [hunk 1]
```diff
@@ -14,6 +14,9 @@ pub mod sector;
 pub mod state_tree;
 pub mod version;
 
+use fil_actors_runtime_v9::builtin::network;
+pub const EPOCHS_IN_DAY: fvm_shared::clock::ChainEpoch = network::EPOCHS_IN_DAY;
```
Comment (lemmih) [inline utils/forest_shim/src/lib.rs:18]: Let's move this into the `clock` module to match `fvm_shared`.

File: utils/forest_shim/src/clock.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use fil_actors_runtime_v9::builtin::network;
+pub const EPOCHS_IN_DAY: fvm_shared::clock::ChainEpoch = network::EPOCHS_IN_DAY;
```
Comment (lemmih) [inline utils/forest_shim/src/clock.rs:5]: Actually, let's hoist out the constant and define it ourselves.

Comment (sudo-shashank) [reply utils/forest_shim/src/clock.rs:5]: okay

File: Cargo.toml [hunk 3]
```diff
@@ -145,37 +144,36 @@ unsigned-varint = { version = "0.7", default-features = false }
 url = { version = "2.3", features = ["serde"] }
 which = "4.3"
 
-fil_actor_account_v10 = { git = "https://github.com/ChainSafe/fil-actor-states" }
-fil_actor_account_v8 = { git = "https://github.com/ChainSafe/fil-actor-states" }
+# fil_actor_account_v10 = { git = "https://github.com/ChainSafe/fil-actor-states" }
+# fil_actor_account_v8 = { git = "https://github.com/ChainSafe/fil-actor-states" }
```
Comment (lemmih) [inline Cargo.toml:148]: These can just be deleted.

**Review by lemmih — APPROVED** [review]
Looks good.

File: utils/forest_shim/src/clock.rs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+/// Duration of each tipset epoch.
+pub const EPOCH_DURATION_SECONDS: i64 = 30;
```
Comment (LesnyRumcajs) [inline utils/forest_shim/src/clock.rs:5]: why do we redefine this? It should be available in the FVM.

Comment (lemmih) [reply utils/forest_shim/src/clock.rs:5]: I vote use what we can from the FVM but don't use anything from the actor runtimes.

Comment (LesnyRumcajs) [reply utils/forest_shim/src/clock.rs:5]: @lemmih why? We are using the entire `Policy` struct from [here](https://github.com/filecoin-project/builtin-actors/blob/master/runtime/src/runtime/policy.rs)

Comment (lemmih) [reply utils/forest_shim/src/clock.rs:5]: Indeed, and that has been a nightmare. A new copy of the `Policy` is pushed every network upgrade. We need a proper way of dealing with that. Right now, we're just kicking the can down the road.

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**